src/models/product.ts

```diff
@@ -0,0 +1,10 @@
+import {Document} from "mongoose";
+
+export interface Product {
```

Comment:
why not interface Product extends document? and then in controllers: " const products: Product[] = AWAIT queries.find()"